libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationTest.java

```diff
@@ -28,199 +28,257 @@
     Milestone milestone = new StepMilestone.Builder().build();
+
     navigation.addMilestone(milestone);
+
     assertTrue(navigation.getMilestones().contains(milestone));
   }
 
   @Test
   public void addMilestone_milestoneOnlyGetsAddedOnce() throws Exception {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder().defaultMilestonesEnabled(false).build();
-    MapboxNavigation navigationWithOptions = new MapboxNavigation(mock(Context.class),
-      ACCESS_TOKEN, options, mock(NavigationTelemetry.class), mock(LocationEngine.class));
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
+
     Milestone milestone = new StepMilestone.Builder().build();
     navigationWithOptions.addMilestone(milestone);
     navigationWithOptions.addMilestone(milestone);
+
     assertEquals(1, navigationWithOptions.getMilestones().size());
   }
 
   @Test
   public void removeMilestone_milestoneDidGetRemoved() throws Exception {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder().defaultMilestonesEnabled(false).build();
-    MapboxNavigation navigationWithOptions = new MapboxNavigation(mock(Context.class),
-      ACCESS_TOKEN, options, mock(NavigationTelemetry.class), mock(LocationEngine.class));
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
+
     Milestone milestone = new StepMilestone.Builder().build();
     navigationWithOptions.addMilestone(milestone);
-    assertEquals(1, navigationWithOptions.getMilestones().size());
     navigationWithOptions.removeMilestone(milestone);
+
     assertEquals(0, navigationWithOptions.getMilestones().size());
   }
 
   @Test
   public void removeMilestone_milestoneDoesNotExist() throws Exception {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder().defaultMilestonesEnabled(false).build();
-    MapboxNavigation navigationWithOptions = new MapboxNavigation(mock(Context.class),
-      ACCESS_TOKEN, options, mock(NavigationTelemetry.class), mock(LocationEngine.class));
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
+
     Milestone milestone = new StepMilestone.Builder().build();
     navigationWithOptions.addMilestone(new StepMilestone.Builder().build());
     navigationWithOptions.removeMilestone(milestone);
+
     assertEquals(1, navigationWithOptions.getMilestones().size());
   }
 
   @Test
   public void removeMilestone_nullRemovesAllMilestones() throws Exception {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder().defaultMilestonesEnabled(false).build();
-    MapboxNavigation navigationWithOptions = new MapboxNavigation(mock(Context.class),
-      ACCESS_TOKEN, options, mock(NavigationTelemetry.class), mock(LocationEngine.class));
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
     navigationWithOptions.addMilestone(new StepMilestone.Builder().build());
```

Comment:
We could extract this block of code into a `private` method called `addABunchOfMilestones` for example. It'd reflect better our intention and it'd be clearer and easier to understand.